ktor-server/ktor-server-test-host/jvm/src/io/ktor/server/testing/BaseTestJvm.kt

```diff
@@ -5,31 +5,42 @@
 
 package io.ktor.server.testing
 
+import io.ktor.junit.*
 import kotlinx.coroutines.*
-import kotlinx.coroutines.debug.junit4.*
-import org.junit.*
-import org.junit.rules.*
+import kotlinx.coroutines.debug.junit5.*
+import org.junit.jupiter.api.*
+import java.lang.reflect.*
+import java.util.*
 import kotlin.time.*
 import kotlin.time.Duration.Companion.seconds
 
+@CoroutinesTimeout(5 * 60 * 1000)
+@ErrorCollectorTest
 actual abstract class BaseTest actual constructor() {
```

Comment:
Should we keep `ktor-server-test-host` using junit4 and make a separate module for junit5? Let's discuss

Reply:
The problem here is that it's used in other projects, so we'd have a swath of tests using junit 4.  The only big difference here is that we're going from 60 seconds timeout for each test to 5 minutes total because of how the annotations work.  You can override the annotation on the child classes and methods if another timeout is needed, though it doesn't seem to be a problem atm...

Reply:
lgtm